stdlib/test/builtin/test_int_literal.mojo

```diff
@@ -78,6 +78,11 @@ def test_abs():
     assert_equal(abs(0), 0)
 
 
+def test_indexer():
+    assert_equal(1, (1).__index__())
+    assert_equal(88, (88).__index__())
```

Comment:
Due to implicit conversion traps, with literal (non-materializable) types it's better to force the function to resolve to the desired overload:
```suggestion
    assert_equal(1, IntLiteral.__index__(1))
    assert_equal(88, IntLiteral.__index__(88))
```